src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/ModelAttributes.cs

```diff
@@ -5,27 +5,25 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Metadata;
 
 namespace Microsoft.AspNetCore.Mvc.ModelBinding
 {
     /// <summary>
-    /// Provides access to the  combined list of attributes associated a <see cref="Type"/> or property.
+    /// Provides access to the combined list of attributes associated with a <see cref="Type"/>, property, or parameter.
     /// </summary>
     public class ModelAttributes
     {
+        private static readonly IEnumerable<object> _emptyAttributesCollection = Enumerable.Empty<object>();
+
         /// <summary>
         /// Creates a new <see cref="ModelAttributes"/> for a <see cref="Type"/>.
         /// </summary>
         /// <param name="typeAttributes">The set of attributes for the <see cref="Type"/>.</param>
         public ModelAttributes(IEnumerable<object> typeAttributes)
```

Comment:
It's awkward having these constructor overloads to represent different scenarios, since you might also want to pass a single `IEnumerable<object>` to represent parameter attributes. It would be more explict if we only had public static factory methods. What's the feeling on a breaking change for that?

Reply:
I agree with your feedback.

My suggestion would be:
1. Introduce static factories now
2. Mark constructors obsolete (other than the one with a single list)
3. Update our code to call new constructors
4. File an issue to remove obsoleted APIs in 3.0

Reply:
Done